src/main/resources/org/broadinstitute/hellbender/utils/config/GATKConfig.properties

```diff
@@ -0,0 +1,65 @@
+# ======================================================================================================================
+# System Properties
+#
+# All Configuration options in ths section are to be injected into the JRE
+# as System Properties.
+#
+# Any property beginning with `system.` (without quotes) will be injected into the JRE as a System Property.
+# Properties not starting with `system.` (without quotes) will NOT be used as System Properties.
+#=======================================================================================================================
+
+# ---------------------------------------------------------------------------------
+# General Options:
+# ---------------------------------------------------------------------------------
+
+gatk_stacktrace_on_user_exception = true
+
+# ---------------------------------------------------------------------------------
+# SAMJDK Options:
+# ---------------------------------------------------------------------------------
+
+system.samjdk.use_async_io_read_samtools = true
+system.samjdk.use_async_io_write_samtools = true
+system.samjdk.use_async_io_write_tribble = false
+system.samjdk.compression_level = 1
+
+# ----------------------------------------------------------
+# Spark Options:
+# ----------------------------------------------------------
+
+spark.kryoserializer.buffer.max = 512m
+spark.io.compression.codec = lzf
+spark.yarn.executor.memoryOverhead = 600
+
+spark.driver.maxResultSize = 0
+spark.driver.userClassPathFirst = true
+spark.driver.extraJavaOptions =
+
+spark.executor.extraJavaOptions =
+
+# ---------------------------------------------------------------------------------
+# Other Options:
+# ---------------------------------------------------------------------------------
+
+snappy.disable = true
+
+# ======================================================================================================================
+# Tool Configuration File
+#
+# Contains config options for various pieces of the GATK
+# ======================================================================================================================
+
+# ---------------------------------------------------------------------------------
+# General Options:
+# ---------------------------------------------------------------------------------
+
+codec_packages = htsjdk.variant      ,            htsjdk.tribble         ,         org.broadinstitute.hellbender.utils.codecs          
```

Comment:
I would like to remove the three libraries as a developer, but not like an user (so command-line overrides are not a possibility). I am not suggesting to remove the default configuration file, but have a way to provide this defaults in my GATK-based toolkit in a simpler way without requiring the user to provide a file, and it is still not clear for me how to do that.

In the case of the config-file format, for me readability is very important because if not it could lead to really bad configuration due to a complex set of properties. For the case that I reported in this comment, the solution is not only adding a `codec_excludes`: it will require to have the following properties:
* `codec_packages`: packages to search for the codecs, and include all of them (except excluded)
* `codec_classes`: single-classes to include for codecs, in any package because full qualifiers are provided.
* `codec_packages_excludes`: exclude the following packages, to allow removal of nested packages specified on the `codec_packages`
* `codec_classes_excludes`: exclude the following classes, to allow removal of concrete classes found in the packages specified on the `codec_packages`

This is assuming that the classes/packages are in different lists (if not, it can be a problem with parsing). In that case, the example config file that I showed before as YML will become the following:

```properties.java
codec_packages: htsjdk.tribble, org.broadinstitute.hellbender.utils.codecs.samtools, org.magicdgs.htsjdk.codecs
codec_packages_exclude: htsjdk.variant
codec_classes_exclude: htsjdk.tribble.bed.BEDCodec
codec_classes: org.magicdgs.beta.codecs.GTFCodec
```

Which is more compact, but it is difficult to know which codecs are included at the end of applying the configuration for each of the packages.